kernel/exports/nmmgr.c

```diff
@@ -34,12 +34,17 @@ static nmmgr_list_t nmmgr_handlers;
 
 /* Locate a name handler for a given path name */
 nmmgr_handler_t * nmmgr_lookup(const char *fn) {
-    nmmgr_handler_t *cur;
-
-    /* Scan the handler table and look for a path match */
-    LIST_FOREACH(cur, &nmmgr_handlers, list_ent) {
-        if(!strncasecmp(cur->pathname, fn, strlen(cur->pathname)))
-            break;
+    nmmgr_handler_t *cur, *tmp;
+    size_t          cur_len = 0;
+
+    /* Scan the handler table and look for the best path match */
+    LIST_FOREACH(tmp, &nmmgr_handlers, list_ent) {
+        if(!strncasecmp(tmp->pathname, fn, strlen(tmp->pathname))) {
```

Comment:
Add another size_t local variable to hold the strlen(tmp->pathname) value so you don't have to recalculate it multiple times.

Reply:
I was figuring compiler optimization would be able to recognize that, but good call.